src/mono/wasm/runtime/http.ts

```diff
@@ -65,80 +65,32 @@ export function http_wasm_abort_response(res: ResponseExtension): void {
     }
 }
 
-export function http_wasm_readable_stream_controller_enqueue(pull_state: PullStateExtension, bufferPtr: VoidPtr, bufferLength: number): void {
-    const controller = pull_state.__controller;
-    const pull_promise_control = pull_state.__pull_promise_control;
-    mono_assert(controller, "expected controller");
-    mono_assert(pull_promise_control, "expected pull_promise_control");
-    try {
-        if (bufferLength === 0) {
-            controller.close();
-            pull_state.dispose();
-            pull_state.__pull_promise_control = null;
-            pull_state.__controller = null;
-        } else {
-            // the bufferPtr is pinned by the caller
-            const view = new Span(bufferPtr, bufferLength, MemoryViewType.Byte);
-            // because https://github.com/WebAssembly/design/issues/1162 we need to copy the buffer
-            // also it doesn't make much sense to use byob
-            const copy = view.slice() as Uint8Array;
-            controller.enqueue(copy);
-        }
-        pull_promise_control.resolve();
-    }
-    catch (err) {
-        pull_state.dispose();
-        pull_promise_control.reject(err);
-    }
-    finally {
-        pull_state.__pull_promise_control = null;
-    }
-}
-
-export function http_wasm_readable_stream_controller_error(pull_state: PullStateExtension, error: Error): void {
-    const controller = pull_state.__controller;
-    mono_assert(controller, "expected controller");
-    pull_state.__pull_promise_control?.reject(error);
-    pull_state.__fetch_promise_control?.reject(error);
-    pull_state.dispose();
-    controller.error(error);
-    pull_state.__pull_promise_control = null;
+export async function http_wasm_request_stream_write(ts: TransformStreamExtension, bufferPtr: VoidPtr, bufferLength: number): Promise<void> {
+    mono_assert(bufferLength > 0, "expected bufferLength > 0");
+    await ts.__writer.ready;
+    // the bufferPtr is pinned by the caller
+    const view = new Span(bufferPtr, bufferLength, MemoryViewType.Byte);
+    const copy = view.slice() as Uint8Array;
+    await ts.__writer.write(copy);
```

Comment:
I would love this to be cancellable.

Reply:
I assume this may throw when
- TPC breaks or server kills the connection mid flight
- we triggered AbortController

Reply:
Unfortunately not, I need to race all of these with the `fetch_promise_control` e.g. 
```js
await Promise.race([ts.__writer.write(copy), ts.__fetch_promise_control.promise]);
```
or do you have another idea?

```js
var t = new TransformStream();
var w = t.writable.getWriter();
var a = new AbortController();

var f = fetch("", {
    method: "POST",
    body: t.readable,
    duplex: "half",
    signal: a.signal
});

a.abort();

var wp = w.write(new Uint8Array([1, 2, 3]));

var done = Promise.race([wp, f]);

// wp is pending
// done / f is rejected
```

Reply:
I opened a bug report in chrome: https://bugs.chromium.org/p/chromium/issues/detail?id=1480250
Because I would expect that any fetch error would properly cancel the `ReadableStream` (which the browser locks) and thus any writer promises would be rejected.